src/App.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import Navbar from './components/header';
+import Home from './components/homepage';
+import Details from './components/detailpage';
+import { fetchCoins } from './redux/reducer/currencies';
+
+const App = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchCoins());
+  }, [dispatch]);
+
+  return (
+    <div className="App">
+      <Navbar />
+      <div>
+        <Routes>
+          <Route path="/" element={<Home />} />
+          <Route path="/display" element={<Details />} />
```

Comment:
- [ ] Your route for all the coins are the same, `/display`. But they should be unique.

  An example if we navigate to the Ethereum coin page, we should have something like this: `/coins/ETH`. This is a hint of what the route should look like.

  ```
  /coins/:id
  ```
  Kindly refactor your routes so that all the coins have their own unique routes. You can consider looking at **`useParams`** from the `react-router-dom` package, and using the URL id to filter or fetch the coin

  ![image](https://user-images.githubusercontent.com/54848363/210766497-afbb8070-95a7-4d5e-8943-4fbe93481909.png)